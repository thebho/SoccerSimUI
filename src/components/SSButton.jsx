// @flow
import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';

type Props = {
  label: string,
  onClick: Function,
}

const SSButton = (props: Props) => (
  <RaisedButton
    label={props.label}
    onClick={props.onClick}
    backgroundColor="#C5E1A5"
    style={{
      margin: 10,
    }}
  />
)

export default SSButton;
