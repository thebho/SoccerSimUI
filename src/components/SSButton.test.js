import React from 'react';
import ReactDOM from 'react-dom';
import { fn } from 'jest';
import SSButton from './SSButton';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import RaisedButton from 'material-ui/RaisedButton';

describe('SSButton', () => {
  it('matches snapshot', () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(
      <SSButton label='Test Label' onClick={onClickMock} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.simulate('click');
    expect(onClickMock).toHaveBeenCalled();
  });
});
