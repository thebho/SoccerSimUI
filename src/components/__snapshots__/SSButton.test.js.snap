// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSButton matches snapshot 1`] = `
<RaisedButton
  backgroundColor="#C5E1A5"
  disabled={false}
  fullWidth={false}
  label="Test Label"
  labelPosition="after"
  onClick={[Function]}
  primary={false}
  secondary={false}
  style={
    Object {
      "margin": 10,
    }
  }
/>
`;
